components/director/internal/tenantmapping/mapper_for_system_auth.go

```diff
@@ -0,0 +1,109 @@
+package tenantmapping
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/systemauth"
+	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
+	"github.com/pkg/errors"
+)
+
+func NewMapperForSystemAuth(systemAuthSvc systemauth.SystemAuthService, scopesGetter ScopesGetter) *mapperForSystemAuth {
+	return &mapperForSystemAuth{
+		systemAuthSvc: systemAuthSvc,
+		scopesGetter:  scopesGetter,
+	}
+}
+
+type mapperForSystemAuth struct {
+	systemAuthSvc systemauth.SystemAuthService
+	scopesGetter  ScopesGetter
+}
+
+func (m *mapperForSystemAuth) GetTenantAndScopes(ctx context.Context, reqData ReqData, authID string, authFlow AuthFlow) (string, string, error) {
+	sysAuth, err := m.systemAuthSvc.GetGlobal(ctx, authID)
+	if err != nil {
+		return "", "", errors.Wrap(err, "while retrieving system auth from database")
+	}
+
+	refObjType, err := sysAuth.GetReferenceObjectType()
+	if err != nil {
+		return "", "", errors.Wrap(err, "while getting reference object type")
+	}
+
+	var tenantAndScopesFunc func(sysAuth *model.SystemAuth, refObjType model.SystemAuthReferenceObjectType, reqData ReqData, authFlow AuthFlow) (string, string, error)
```

Comment:
Suggestion: I understand what you wanted to achieve - not duplicating passing parameters. However, I'd say that using  simple switch/case with passing the parameters would be easier to read.

```go
return m.getTenantAndScopesForIntegrationSystem(sysAuth, refObjType, reqData, authFlow)
```

Reply:
modified